lib/model/wallet.js

```diff
@@ -107,20 +108,19 @@ Wallet.prototype.addCopayer = function(copayer) {
   this._updatePublicKeyRing();
 };
 
-Wallet.prototype.updateCopayerRequestKey = function(copayerId, requestPubKey, signature) {
+Wallet.prototype.addCopayerRequestKey = function(copayerId, requestPubKey, signature, restrictions, name) {
   $.checkState(this.copayers.length == this.n);
 
-  var c = _.find(this.copayers, {
-    id: copayerId,
-  });
-
-  $.checkState(c)
-    .checkState(c.isTemporaryRequestKey);
+  var c = this.getCopayer(copayerId);
 
-  c.requestPubKey = requestPubKey;
-  c.isTemporaryRequestKey = false;
-  c.signature = signature;
-  this._updatePublicKeyRing();
+  //new ones go first
+  c.requestPubKeys.unshift({
```

Comment:
So, what about the first key (the one used to create the copayer). Will it be different in that it won't have additional attributes?